modules/programs/tarsnap.nix

```diff
@@ -0,0 +1,300 @@
+          Note that exclusions specified with exclude take precedence over
+          inclusions. If no inclusions are explicitly specified, all entries are
+          processed by default. The --include option is especially useful when
+          filtering archives. For example, the command <command>tarsnap -c -f
+          foo-backup --include='*foo*' @@all-backup</command> creates a new
+          archive foo-backup containing only the entries from all-backup
+          containing the string Sq foo.
+        '';
+      };
+
+      insaneFilesystems = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Allow descent into synthetic filesystems such as procfs. Normally
+          archiving of such filesystems is a silly thing to do, hence the name
+          of the option.
+        '';
+      };
+
+      isoDates = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Print file and directory dates as yyyy-mm-dd hh:mm:ss.
+        '';
+      };
+
+      keyfile = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = ''
+          Obtain encryption, authentication, and access keys from key-file. This
+          file should have been generated by
+          <command>tarsnap-keygen(1)</command>.
+        '';
+      };
+
+      memoryUsage = mkOption {
+        type = types.enum [ "normal" "low" "verylow" ];
+        default = "normal";
+        description = ''
+          Set the memory-usage mode of tarsnap.
+
+          For low, reduce memory usage by not caching small files. This may be
+          useful when backing up files of average size less than 1 MB if the
+          available RAM in kilobytes is less than the number of files being
+          backed up.
+
+          For verylow reduce memory usage, by approximately a factor of 2 beyond
+          the memory usage when --lowmem is specified, by not caching anything.
+        '';
+      };
+
+      maxbw = mkOption {
+        type = types.nullOr types.int;
+        default = null;
+        description = ''
+          Interrupt archival if more than numbytes bytes of upstream bandwidth
+          is used
+        '';
+      };
+
+      maxbwRate = mkOption {
+        type = types.nullOr types.int;
+        default = null;
+        description = ''
+          Limit download and upload bandwidth used to maxbwRate bytes per
+          second.
+        '';
+      };
+
+      maxbwRateDown = mkOption {
+        type = types.nullOr types.int;
+        default = null;
+        description = ''
+          Limit download bandwidth used to maxbwRate bytes per second.
+        '';
+      };
+
+      maxbwRateUp = mkOption {
+        type = types.nullOr types.int;
+        default = null;
+        description = ''
+          Limit upload bandwidth used to maxbwRate bytes per second.
+        '';
+      };
+
+      nodump = mkOption {
```

Comment:
The description doesn't seem to match the option. Also best avoid negated option names since it unintuitively requires you to set it to false to enable it.